IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,156 @@
+//Coded by James Sedlacek
+//Date: 05/15/2020
+
+//Create a struct called Flavor. Add the following properties. Think about what type they should be:
+// name
+// rating
+struct Flavor {
+    var name: String
+    var rating: Int
+}
+
+//Create an enum called Size.
+//Give it a case for small, medium, and large.
+//Make the enum have Double raw values. The raw value should represent the size's price. For example, small might be 3.99.
+
+enum Size: Double {
+    case small = 3.99
+    case medium = 4.99
+    case large = 5.99
+}
+
+//Create a struct called Cone.
+//Give the struct properties (constants) that hold a flavor, topping (as a String), and size.
+//Create a function called eat. It shouldn't have any arguments. It should print a string of text that includes the name of the cone's flavor. For example: "Mmm! I love !"
+
+struct Cone {
+    var flavor: Flavor
+    var topping: String?
+    var size: Size
+    
+    func eat() {
+        print("Mmm! I love \(flavor.name)!")
+    }
+}
+
+//Create a class called IceCreamShop.
+//An ice cream shop needs a menu, so create variables that hold the various flavors, sizes, and toppings you want to have in your shop.
+//Create a variable called totalSales. This will be used to keep track of how much money was made.
+
+class IceCreamShop {
+    var totalSales: Double
+    var flavors: [Flavor]
+    var sizes: [Size]
+    var toppings: [String]
+    
+    init(flavors: [Flavor], toppings: [String], sizes: [Size]){
+        self.totalSales = 0.0
+        self.flavors = flavors
+        self.toppings = toppings
+        self.sizes = sizes
+    }
+    
+    /*
+     If a customer asks which flavors are available, we need to be ready to tell them.
+     In the function, build a string that lists out the names of each flavor in the shop's flavors array whose rating is over 4.0. For example, the string might say "Our top flavors are ." Hint: You may want to loop through the array of flavors in order to access each one's name.
+     When the string has been created, print it.
+     */
+    func listTopFlavors() {
+        var message: String = "Our top flavors are "
+        var topFlavors: [String] = []
+        var totalTopFlavors: Int = 0
+        for flavor in self.flavors {
+            if flavor.rating > 4 {
+                topFlavors.append(flavor.name)
+                totalTopFlavors += 1
+            }
+        }
+        
+        //finishing the last flavor with "and". For example,", , and ", instead of just ", , "
+        var index: Int = 0
+        for f in topFlavors {
+            message += f + ", "
+            if index == topFlavors.count - 2 {
+                message += "and "
+            }
```

Comment:
Nice job figuring out the formatting for the commas and "and" in-between the flavors! However, notice that your print out in the console adds a trailing comma at the end of the sentence instead of a period:

> Our top flavors are Vanilla, Chocolate, and Mint, 

Be sure to try to consider edge cases. For example, if there were only two top flavors currently the comma should be eliminated:

> Our top flavors are Vanilla, and Chocolate, 

Also, if there is only one top flavor you should make "flavors" singular and us "is" instead of "are"

> Our top flavors are Vanilla,